21.md

```diff
@@ -0,0 +1,210 @@
+
+The first request is unchanged from the base specification.
+
+### Service-side first response
+
+`SERVICE` must alter its JSON response to the first callback to include a `currencies` field, as follows:
+
+```typescript
+type BaseResponse = {
+    tag: "payRequest",
+    metadata: string,
+    callback: string,
+    maxSendable: number,
+    minSendable: number
+}
+
+type Currency = {
+  code: string, // Code of the currency, used as an ID for it. E.g.: BRL
+  name: string, // Name of the currency. E.g.: Reais
+  symbol: string, // Symbol of the currency. E.g.: R$
+  decimals: number, // Integer; Number of decimal places. E.g.: 2
+  multiplier: number, // Double; Number of millisatoshis per smallets unit of currency. E.g.: 5405.405
+  convertible?: bool // Whether the currency can be converted into.
+}
+
+type ExtendedResponse = BaseResponse & {
+  currencies: Currency[]
+}
+```
+
+```diff
+{
+  "tag": "payRequest",
+  "metadata": '[["text/plain","$kenu ⚡ bipa.app"]]',
+  "callback": "https://api.bipa.app/ln/request/invoice/kenu",
+  "maxSendable": 1000000000,
+  "minSendable": 1000,
++ "currencies": [
++   {
++     "code": "BRL",
++     "name": "Real",
++     "symbol": "R$",
++     "decimals": 2,
++     "multiplier": 5404.405,
++     "convertible": true
++   }
++ ]
+}
+```
+
+- The inclusion of the `currencies` field implies the support of this extension
+- The inclusion of a `currency` implies it can be used for denomination of an amount
+- The inclusion of a `convertible currency` implies the `SERVICE` can quote and guarantee a price for given currency
+- The `multiplier` returned on this response is not guaranteed by the `SERVICE` and is subject to change
+- The `code` of a `currency` will be used as an identifier for the next request and must be unique
+- The `code` and other information must be according to [ISO-4217](https://en.wikipedia.org/wiki/ISO_4217) for currencies that are included in it
+- The order of the `currencies` must be interpreted by the `WALLET` as the receiving user preference for a currency
+
+### Wallet-side second request
+
+Upon receiving the `currencies` field on the response, the `WALLET` shows the user it has the option of denominating the amount in one of the `currencies` or for the payment to be credit as the different currency for the receiver. If the BTC being used for the payment is was bought as part of the payment, the `WALLET` may inform the `SERVICE` about it's cost.
+
+The inputs that must be gathered from the user are:
+- An optional denominating currency and amount (`CURRENCY_D` and `AMOUNT_D`)
+- An optional source currency and amount (`CURRENCY_S` and `AMOUNT_S`)
+- An optional target currency (`CURRENCY_T`)
+
+The most general case has all the possible parameters present.
+It will generate an invoice with amount equivalent to `AMOUNT_D` `CURRENCY_D`, that will be converted into `CURRENCY_T` by the `SERVICE` upon payment, which will be done with BTC that costed the sender `AMOUNT_S` `CURRENCY_S`.
```

Comment:
```suggestion
It will generate an invoice with the amount equivalent to `AMOUNT_D` `CURRENCY_D`, which will be converted into `CURRENCY_T` by the `SERVICE` upon payment, which will be done with BTC that costs the sender `AMOUNT_S` `CURRENCY_S`.
```